cxx/distributions/nonconjugate.hh

```diff
@@ -42,4 +46,24 @@ class NonconjugateDistribution : public Distribution<T> {
     }
     return score;
   }
+
+  // Transition the current latent values using Metropolis-Hastings.
+  // Children classes are welcome to replace this with something more
+  // powerful (like Hamiltonian Monte Carlo) if they like.
+  virtual void transition_theta(std::mt19937* prng) {
+    std::vector<double> old_latents = store_latents();
```

Comment:
Should we turn these in to pointers (and init_theta returns a pointer)? I guess I'm asking in case the distributions have a sizeable latent space (I'm thinking string related) and so copying vectors around becomes expensive.

Reply:
My impression is that in this sort of context, the compiler will apply named return value optimization which gets rid of the copies.  See for example https://stackoverflow.com/questions/12604431/returning-an-stl-vector-from-a-function-copy-cost